app/services/api/webflow-helpers.ts

```diff
@@ -0,0 +1,182 @@
+import { Schedule } from "../../screens"
+import { ScheduleCardProps } from "../../screens/ScheduleScreen/ScheduleCard"
+import { useSchedule, useWorkshops } from "./webflow-api"
+import {
+  CustomScheduleProps,
+  CustomSpeakerProps,
+  CustomWorkshopProps,
+  ScheduleProps,
+  SpeakerProps,
+  WorkshopProps,
+} from "./webflow-api.types"
+
+/*
+ * Converting workshop data from "type ids" to "type names"
+ */
+export const cleanedWorkshops = (
+  workshopsData?: CustomWorkshopProps[],
+  speakersData?: SpeakerProps[],
+): WorkshopProps[] => {
+  return workshopsData
+    ?.filter((workshop) => !workshop._archived && !workshop._draft)
+    .map((workshop) => ({
+      ...workshop,
+      level:
+        workshop.level === "bcb33aac3cd85ef6f2e7a97cf23c9771"
+          ? "Beginner"
+          : workshop.level === "e9d1df0d23f4049bd9d1a6fe83c5db01"
+          ? "Intermediate"
+          : workshop.level === "860319fadc9cd03654561fba21490285"
+          ? "Advanced"
+          : undefined,
+      type:
+        workshop.type === "d9770c43cd59f01f2d60b288d65c1f90"
+          ? "New"
+          : workshop.type === "c8af4236d64f5c25c09e61e4633badb0"
+          ? "Top Seller"
+          : undefined,
+      "instructor-info": speakersData?.find(
+        (speaker) => speaker._id === workshop["instructor-info"],
+      ),
+      "second-instructor-2": speakersData?.find(
+        (speaker) => speaker._id === workshop["second-instructor-2"],
+      ),
+    }))
+}
+
+/*
+ * Converting speakers data from "type ids" to "type names"
+ */
+export const cleanedSpeakers = (speakersData?: CustomSpeakerProps[]): SpeakerProps[] => {
+  return speakersData?.map((speaker) => ({
+    ...speaker,
+    "speaker-type":
+      speaker["speaker-type"] === "97dae28f90a767132ee88e80a8537af8"
+        ? "Speaker"
+        : speaker["speaker-type"] === "079e51435c82a91426f9c3acc7b0343a"
+        ? "Panelist"
+        : speaker["speaker-type"] === "f23ef92d0cef6be6fd60654d54770c96"
+        ? "Workshop"
+        : speaker["speaker-type"] === "07948ce9361d13f707fdb4e663cbe9a5"
+        ? "Emcee"
+        : undefined,
+    "talk-level":
+      speaker["talk-level"] === "2f3097a3529a99ed4d688e9ce05034d6"
+        ? "Beginner"
+        : speaker["talk-level"] === "33984dd1db455114d65e3bd9989f4fad"
+        ? "Intermediate"
+        : speaker["talk-level"] === "ce1ba34575f5a7e30ba9c4f3c33c8211"
+        ? "Advanced"
+        : undefined,
+  }))
+}
+
+/*
+ * Converting schedule data from "type ids" to "type names"
+ */
+export const cleanedSchedule = (
+  scheduleData?: CustomScheduleProps[],
+  speakersData?: SpeakerProps[],
+): ScheduleProps[] => {
+  return scheduleData
+    ?.filter((schedule) => !schedule._archived && !schedule._draft)
+    .map((schedule) => ({
+      ...schedule,
+      day:
```

Comment:
To make this easier to read, we could consider doing something like:

```
const DAY_MAP = {
  63ac4ade8b2d5a981780570e01bed34d: 'Wednesday',
  ed2cfa99e27dce5d1a425a419f170eb3: 'Thursday',
  whateverFridayIs: 'Friday',
}

const TYPE_MAP = {
  976a2eee2ab173440affe93d0a20bf4d: 'Talk',
  c728019fbbce26cc72918695f31e7d4d: 'Lightning Talk',
  2e399bc3e027ec007832e20e47c5b83b: 'Break',
  b2f17244cf6bd0782c2a099568169219: 'Friday',
}

// ...

.map((schedule) => ({
      ...schedule,
      day: DAY_MAP[schedule.day],
      type: TYPE_MAP[schedule.type],
```

EDIT: Since there are a few loops like this, we could ultimately create an object mapping structure similar to how it works for i18n

```
const WEBFLOW_MAP = {
  speakers: {
   97dae28f90a767132ee88e80a8537af8: 'Speaker',
  079e51435c82a91426f9c3acc7b0343a: 'Panelist',
  // ...
 },
 days: {
  63ac4ade8b2d5a981780570e01bed34d: 'Wednesday',
  ed2cfa99e27dce5d1a425a419f170eb3: 'Thursday',
  whateverFridayIs: 'Friday',
 }
}

// ...

WORKFLOW_MAP.speakers[guid]
WORKFLOW_MAP.days[guid]

// etc...
```

Even collection UUIDs could be stored here I'd think

Reply:
Great idea! fixed in latest push